sklearn/metrics/classification.py

```diff
@@ -292,6 +293,204 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None):
     return CM
 
 
+def multilabel_confusion_matrix(y_true, y_pred, sample_weight=None,
+                                labels=None, samplewise=False):
+    """Compute multi-label confusion matrix label-wisely (default) or
+    sample-wisely (samplewise=True) to evaluate the accuracy of a
+    classification, and output confusion matrices for each label or sample
+
+    In multilabel confusion matrix :math:`MCM`, the count of true negatives
+    is :math:`MCM_{:,0,0}`, false negatives is :math:`MCM_{:,1,0}`,
+    true positives is :math:`MCM_{:,1,1}` and false positives is
+    :math:`MCM_{:,0,1}`.
+
+    Multiclass and multilabel-indicator tasks will be treated as if
+    binarized under a one-vs-rest transformation. Returned confusion
+    matrices will be in the order of sorted unique labels in the union
+    of (y_true, y_pred).
+
+    Parameters
+    ----------
+    y_true : 1d array-like, or label indicator array / sparse matrix
+        of shape (n_samples, n_outputs) or (n_samples,)
+        Ground truth (correct) target values.
+
+    y_pred : 1d array-like, or label indicator array / sparse matrix
+        of shape (n_samples, n_outputs) or (n_samples,)
+        Estimated targets as returned by a classifier
+
+    sample_weight : array-like of shape = [n_samples], optional
+        Sample weights
+
+    labels : array-like
+        A list of classes or column indices to select some (or to force
+        inclusion of classes absent from the data)
+
+    samplewise : bool, default=False
+        In the multilabel case, this calculates a confusion matrix per sample
+
+    Returns
+    -------
+    multi_confusion : array, shape (n_outputs, 2, 2)
+        A 2x2 confusion matrix corresponding to each output in the input.
+        When calculating label-wise multi_confusion (default), then
+        n_outputs = n_labels; when calculating sample-wise multi_confusion
+        (samplewise=True), n_outputs = n_samples.
+
+    See also
+    --------
+    confusion_matrix
+
+    Notes
+    -----
+    The multilabel_confusion_matrix calculates label-wise or sample-wise
+    multilabel confusion matrices, labels are binarized under a one-vs-rest
+    way; while confusion_matrix calculates one confusion matrix for confusion
+    between every two labels.
+
+    Examples
+    --------
+    Multilabel-indicator case:
+    >>> from sklearn.metrics import multilabel_confusion_matrix
+    >>> y_true = [[1, 0, 1], [0, 1, 0]]
+    >>> y_pred = [[1, 0, 0], [0, 1, 1]]
+    >>> multilabel_confusion_matrix(y_true, y_pred, samplewise=True)
```

Comment:
why samplewise?

Reply:
This is an example of using samplewise, do you suggest that I replace it with the same data using samplewise=False or just add this example?

Reply:
I think the primary usage is not samplewise, so this should be illustrated first.